vibrato/src/dictionary/connector/raw_connector.rs

```diff
@@ -0,0 +1,249 @@
+mod scorer;
+
+use std::io::{prelude::*, BufReader, Read};
+
+use bincode::{Decode, Encode};
+use hashbrown::HashMap;
+
+use crate::dictionary::connector::raw_connector::scorer::{Scorer, ScorerBuilder};
+use crate::dictionary::connector::{Connector, ConnectorCost};
+use crate::dictionary::mapper::ConnIdMapper;
+use crate::errors::{Result, VibratoError};
+use crate::utils;
+
+const INVALID_FEATURE_ID: u32 = u32::MAX;
+
+#[derive(Decode, Encode)]
+pub struct RawConnector {
+    right_ids: Vec<u32>,
+    left_ids: Vec<u32>,
+    col_size: usize,
+    scorer: Scorer,
+}
+
+impl RawConnector {
+    pub fn new(right_ids: Vec<u32>, left_ids: Vec<u32>, col_size: usize, scorer: Scorer) -> Self {
+        Self {
+            right_ids,
+            left_ids,
+            col_size,
+            scorer,
+        }
+    }
+
+    /// Creates a new instance from `bigram.right`, `bigram.left`, and `bigram.cost`.
+    pub fn from_readers<R, L, C>(right_rdr: R, left_rdr: L, cost_rdr: C) -> Result<Self>
```

Comment:
Could you add small tests for `from_readers`?

Reply:
added